### PR TITLE
 reference/sql: update compatibility description of MySQL

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? 

Update compatibility description of MySQL for reference/sql/mysql-compatibility.md.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [x] This PR is to align with:pingcap/docs-cn#2197
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [ ] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [x] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [x] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/mysql-compatibility.md [hunk 1]
```diff
@@ -102,6 +103,19 @@ In TiDB DDL does not block reads or writes to tables while in operation. However
 + `LOCK [=] {DEFAULT|NONE|SHARED|EXCLUSIVE}`: the syntax is supported, but is not applicable to TiDB. All DDL changes that are supported do not lock the table.
 + `ALGORITHM [=] {DEFAULT|INSTANT|INPLACE|COPY}`: the syntax for `ALGORITHM=INSTANT` and `ALGORITHM=INPLACE` is fully supported, but it works differently from MySQL because some operations that are `INPLACE` in MySQL are `INSTANT` in TiDB. The syntax `ALGORITHM=COPY` is not applicable to TIDB and returns a warning.
 
+Table Option does not support the following syntax:
```
Comment (bb7133) [inline dev/reference/mysql-compatibility.md]:
```suggestion
The following Table Options are not supported in syntax:
```
How about this?

File: dev/reference/mysql-compatibility.md [hunk 2]
```diff
@@ -89,7 +88,9 @@ In TiDB DDL does not block reads or writes to tables while in operation. However
 
 + Add Index:
     - Does not support creating multiple indexes at the same time.
+    - Does not support the `VISIBLE/INVISIBLE` index.
     - Adding an index on a generated column via `ALTER TABLE` is not supported.
+    - Other Index Type (HASH/BTREE/RTREE) is supported in syntax, rather than in function.
```
Comment (bb7133) [inline dev/reference/mysql-compatibility.md]:
```suggestion
    - Other Index Type (HASH/BTREE/RTREE) is supported in syntax, but not applicable.
```
The "Other Index Type"s are just ignored and thus "not applicable".

**Comment by toutdesuite** [comment]
I cannot add tangenta (the original author of the docs-cn PR) as a technical reviewer,  so I add bb7133 (the docs-cn PR reviewer).

File: dev/reference/mysql-compatibility.md [hunk 3]
```diff
@@ -102,6 +103,19 @@ In TiDB DDL does not block reads or writes to tables while in operation. However
...
 + `ALGORITHM [=] {DEFAULT|INSTANT|INPLACE|COPY}`: the syntax for `ALGORITHM=INSTANT` and `ALGORITHM=INPLACE` is fully supported, but it works differently from MySQL because some operations that are `INPLACE` in MySQL are `INSTANT` in TiDB. The syntax `ALGORITHM=COPY` is not applicable to TIDB and returns a warning.
 
+Table Option does not support the following syntax:
+    - `WITH/WITHOUT VALIDATION`
+    - `SECONDARY_LOAD/SECONDARY_UNLOAD`
+    - `CHECK/DROP CHECK`
+    - `STATS_AUTO_RECALC/STATS_SAMPLE_PAGES`
+    - `SECONDARY_ENGINE`
+    - `ENCRYPTION`
+Table Partition does not support the following syntax:
```
Comment (bb7133) [inline dev/reference/mysql-compatibility.md]: Ditto.

Comment (toutdesuite) [reply dev/reference/mysql-compatibility.md]: Thanks a lot.

File: dev/reference/mysql-compatibility.md [hunk 4]
```diff
@@ -102,6 +103,19 @@ In TiDB DDL does not block reads or writes to tables while in operation. However
...
 + `ALGORITHM [=] {DEFAULT|INSTANT|INPLACE|COPY}`: the syntax for `ALGORITHM=INSTANT` and `ALGORITHM=INPLACE` is fully supported, but it works differently from MySQL because some operations that are `INPLACE` in MySQL are `INSTANT` in TiDB. The syntax `ALGORITHM=COPY` is not applicable to TIDB and returns a warning.
 
+The following Table Options are not supported in syntax:
+    - `WITH/WITHOUT VALIDATION`
+    - `SECONDARY_LOAD/SECONDARY_UNLOAD`
+    - `CHECK/DROP CHECK`
+    - `STATS_AUTO_RECALC/STATS_SAMPLE_PAGES`
+    - `SECONDARY_ENGINE`
+    - `ENCRYPTION`
+The following Table Partitions are not supported in syntax:
```
Comment (yikeke) [inline dev/reference/mysql-compatibility.md]:
```suggestion

+ The following Table Partition syntaxes are not supported:
```
The following items are more like "Table Partition syntaxes", not "Table Partitions". Please explain the original Chinese ("Table Partition 不支持以下语法") again if I understand it wrongly. Thanks~ @bb7133

File: dev/reference/mysql-compatibility.md [hunk 5]
```diff
@@ -102,6 +103,19 @@ In TiDB DDL does not block reads or writes to tables while in operation. However
 + `LOCK [=] {DEFAULT|NONE|SHARED|EXCLUSIVE}`: the syntax is supported, but is not applicable to TiDB. All DDL changes that are supported do not lock the table.
 + `ALGORITHM [=] {DEFAULT|INSTANT|INPLACE|COPY}`: the syntax for `ALGORITHM=INSTANT` and `ALGORITHM=INPLACE` is fully supported, but it works differently from MySQL because some operations that are `INPLACE` in MySQL are `INSTANT` in TiDB. The syntax `ALGORITHM=COPY` is not applicable to TIDB and returns a warning.
 
+The following Table Options are not supported in syntax:
```
Comment (yikeke) [inline dev/reference/mysql-compatibility.md]:
```suggestion
+ The following Table Options are not supported in syntax:
```

File: dev/reference/mysql-compatibility.md (hunk 4 above)
Comment (bb7133) [reply dev/reference/mysql-compatibility.md]: You're right, thanks!

**Pull request merged**